File: Arrays and Strings/13. Roman to Integer/Question_3.py
class Solution:
    def romanToInt(self, s: str) -> int:
        sum = 0
        roman_values = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        prev_value = 0
        
        for i in range(len(s)):
            current_value = roman_values[s[i]]
            
            if current_value > prev_value:
                sum += current_value-2*prev_value
            else:
                sum += current_value
            
            prev_value = current_value
        
        return sum
solution = Solution()
s = "MCMXCIV"  
print(solution.romanToInt(s))

File: Arrays and Strings/13. Roman to Integer/Question_3.java
class Solution {
    public int romanToInt(String s) {
        int sum = 0;
        int[] romanValues = new int[26];
        
        romanValues ['I' - 'A'] = 1;
        romanValues ['V' - 'A'] = 5;
        romanValues ['X' - 'A'] = 10;
        romanValues ['L' - 'A'] = 50;
        romanValues ['C' - 'A'] = 100;
        romanValues ['D' - 'A'] = 500;
        romanValues ['M' - 'A'] = 1000;

        int prevValue = 0;

        for(int i=0; i< s.length() ;i++){
            int currentValue = romanValues[s.charAt(i)-'A'];

            if(prevValue < currentValue){
                sum += currentValue -2*prevValue;
            }else{
                sum+= currentValue;
            }
            prevValue = currentValue; 

        }
        return sum;
    }
}
public class Question_3 {
    public static void main(String[] args){
        Solution sol = new Solution();
        String s ="MCMXCIV";
        System.out.println(sol.romanToInt(s));
        System.exit(0);
    }
}
